### DISCUSSION TITLE
Extending Workflow in Joomla 4

### DISCUSSION THREAD

**Discussion by RustyIngles** [joomla/joomla-cms#31431] [2. Cold Features]
### Is your feature request related to a problem? Please describe.

As we all know Joomla 4 is on the horizon and the improvement to workflows are definitely an interesting addition.

From what I've seen, the workflow works great for new content, but what about content that's already live that needs to be edited? 

### Describe the solution you'd like

Could Workflows be used in conjunction with the article version control which set's an edited article to the beginning of the workflow whilst the live version remains live until the edits are approved?

**Comment by bembelimen** [comment]
Hello @RustyIngles one of the basic use cases was a content drafting based on versioning and the workflow. We already have it defined in the [roadmap here](https://www.joomla.org/announcements/release-news/5819-the-day-after-joomla-4-0-planning-the-future.html).

The challenge is currently, that the drafting is not really ready for such an improvement (for the workflow it's easy to build a plugin for it), so someone has to help here.

**Comment by RustyIngles** [comment]
Thanks @bembelimen good to know this is on the roadmap at least.

If we were to build a plugin or at least extend the code for it, presumably this could then be part of the core?

**Comment by bembelimen** [comment]
If the feature works as intented, yes for sure.

If you interested in building it, we can discuss is, if you want via Glip/Email.
